libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -1104,29 +1104,57 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	case FunctionType::Kind::ABIEncode:
 	case FunctionType::Kind::ABIEncodePacked:
 	case FunctionType::Kind::ABIEncodeWithSelector:
+	case FunctionType::Kind::ABIEncodeCall:
 	case FunctionType::Kind::ABIEncodeWithSignature:
 	{
 		bool const isPacked = functionType->kind() == FunctionType::Kind::ABIEncodePacked;
 		solAssert(functionType->padArguments() != isPacked, "");
 		bool const hasSelectorOrSignature =
 			functionType->kind() == FunctionType::Kind::ABIEncodeWithSelector ||
+			functionType->kind() == FunctionType::Kind::ABIEncodeCall ||
 			functionType->kind() == FunctionType::Kind::ABIEncodeWithSignature;
 
 		TypePointers argumentTypes;
 		TypePointers targetTypes;
 		vector<string> argumentVars;
-		for (size_t i = 0; i < arguments.size(); ++i)
+		string selector;
+		vector<ASTPointer<Expression const>> argumentsEncodeFunction;
+
+		if (functionType->kind() == FunctionType::Kind::ABIEncodeCall)
+		{
+			solAssert(arguments.size() == 2, "");
+			auto const tuple = dynamic_cast<TupleExpression const*>(arguments[1].get());
+
+			// Account for tuples with one component which become that component
```

Comment:
I think we might have a helper for this at some point.

Reply:
Actually are you sure you need this special case?
`type(*arguments[1])` will already handle it properly, and I don't see why `IRVariable(*argument)` should have a problem.

Reply:
I am not sure what you're suggesting? That I use/add the tuple (or normal type) directly without iterating over the components in case of a tuple?

Reply:
Yes, nested tuples should be handled properly.